### PR TITLE
[PDI-20194] Variables replacement in PVFS implementation is not worki…

### PR DESCRIPTION
**Pull request by NJtwentyone**
…ng as expected

- caling #initializeVariableSpace earlier in #createFile

please see comment for verification steps: https://hv-eng.atlassian.net/browse/PDI-20194?focusedCommentId=2112669

### CODE REVIEW

**Comment by buildguy** [comment]
[comment]: <> (FrogbotReviewComment)

<div align='center'>

[👍 Frogbot scanned this pull request and did not find any new security issues.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 27m 27s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl core
```
---
:ok_hand: All tests passed!
 
Tests run: **1401**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9747/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
[comment]: <> (FrogbotReviewComment)

<div align='center'>

[👍 Frogbot scanned this pull request and did not find any new security issues.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 22m 32s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl core
```
---
:ok_hand: All tests passed!
 
Tests run: **1401**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9747/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by dcleao — APPROVED** [review]
💯

File: core/src/test/java/org/pentaho/di/connections/common/basic/TestBaseVFSConnectionDetails.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package org.pentaho.di.connections.common.basic;
```
Comment (abryant-hv) [inline core/src/test/java/org/pentaho/di/connections/common/basic/TestBaseVFSConnectionDetails.java:1]: Do we need copyright headers nowadays? I thought so...

File: core/src/test/java/org/pentaho/di/connections/common/basic/TestBaseVFSConnectionDetails.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+import org.pentaho.di.connections.vfs.BaseVFSConnectionDetails;
+import org.pentaho.di.core.variables.VariableSpace;
+
+public abstract class TestBaseVFSConnectionDetails extends BaseVFSConnectionDetails {
+
+  VariableSpace space;
+
+  @Override public VariableSpace getSpace() {
```
Comment (abryant-hv) [inline core/src/test/java/org/pentaho/di/connections/common/basic/TestBaseVFSConnectionDetails.java:10]: Why have the annotations on the same line? Seems very different from most of the codebase (and often makes lines too long)

**Review by abryant-hv — APPROVED** [review]

**Comment by buildguy** [comment]
[comment]: <> (FrogbotReviewComment)

<div align='center'>

[👍 Frogbot scanned this pull request and did not find any new security issues.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 32m 34s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl core
```
---
:ok_hand: All tests passed!
 
Tests run: **1401**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9747/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: core/src/test/java/org/pentaho/di/connections/common/basic/TestBaseVFSConnectionDetails.java (hunk 2 above)
Comment (NJtwentyone) [reply core/src/test/java/org/pentaho/di/connections/common/basic/TestBaseVFSConnectionDetails.java]: addressed with https://github.com/pentaho/pentaho-kettle/commit/900fbd7f11c1f141a27963da6ddde07b0aa42206

**Review by niehusa — APPROVED** [review]

**Pull request merged**